### PR TITLE
SOHO-8147 - Locale.DateFormat functionality to convert Gregorian date…

### PR DESCRIPTION
**Pull request by boehmd1**
…s on demand

**Explain the _details_ for making this change. What existing problem does the pull request solve?**
The hotkey for choosing Today did not work if the DatePicker was not open, got Today date in Gregorian instead of Islamic UmAlQura.

Adding optional argument to LocaleFormatDate attribs JSON.
1. fromGregorian
This should be used by clients who store dates in Gregorian.
If the code is trying to format a date that is Gregorian with a Locale using a non Gregorian calendar, then formatDate will first convert the Gregorian date to  Islamic UmAlQura before formatting.
2. toGregorian
This should be used by clients who store dates in Gregorian.
If the code is trying to format a date that is UmAlQura with a Locale using a non Gregorian calendar, then formatDate will first convert the UmAlQura date to  Gregorian before formatting.

**Related github/jira issue (required)**:
SOHO-8147

**Steps necessary to review your pull request (required)**:
Two functional test were added to test the use of Locale.formatDate using the directives toGregorian and fromGregorian
'Should convert date from Gregorian (if needed) before formatting date (when fromGregorian is true)'
'Should convert date to Gregorian (if needed) before formatting date (when toGregorian is true)'

Also added a page to visually test the methods. test-formadate-conversion.html

**Closing issues**

### CODE REVIEW

File: app/views/components/locale/test-formatdate-conversion.html [hunk 1]
```diff
@@ -0,0 +1,44 @@
+<div class="row">
```
Comment (tmcconechy) [inline app/views/components/locale/test-formatdate-conversion.html]: We probably can remove this visual test if it is covered by a unit test. It doesnt add a whole lot i feel and we are trying to do that to reduce the number of pages to QA.

**Comment by tmcconechy** [comment]
How should i test the "T" key. Tried

http://localhost:4000/components/datepicker/example-index.html?locale=ar-SA
- focus and hit t
But still giving Gregorian.

Are there any other tests i should be doing visually?

**Comment by tmcconechy** [comment]
How should i test the "T" key. Tried

http://localhost:4000/components/datepicker/example-index.html?locale=ar-SA
- focus and hit t
But still giving Gregorian.

Are there any other tests i should be doing visually?

**Comment by boehmd1** [comment]
That seems like the correct steps, at least the steps I user.  Let me try to duplicate that in my sandbox again.  I just merged in changes from your ml.

**Comment by boehmd1** [comment]
Just tested locally, works for me.  I get 1439/10/07

File: app/views/components/locale/test-formatdate-conversion.html (hunk 1 above)
Comment (boehmd1) [reply app/views/components/locale/test-formatdate-conversion.html]: No disagreement.

Comment (tmcconechy) [reply app/views/components/locale/test-formatdate-conversion.html]: ok just delete this file and then push the delete so its gone out of the pr 👍

**Pull request merged**